### PR TITLE
box/lua: new way to define index parts

### LINKED ISSUE #2866: Improve index parts format detection
Having scheme:
```
s = box.schema.create_space('s1', { engine = 'memtx', if_not_exists = true })
i1 = s:create_index('i1', { type = 'tree', parts = {{1, 'unsigned'}}})
```
Running slightly different scripts leads to different results:
```
print(s:create_index('i2', { type = 'tree', parts = {{2, 'unsigned',is_nullable=true}} }).parts[1].is_nullable)
true
```
```
print(s:create_index('i2', { type = 'tree', parts = {2, 'unsigned',is_nullable=true} }).parts[1].is_nullable)
false
```
```is_nullable=true``` is ignored in the second case. The problem is that in lua we have two different styles of parts definition, ```parts = {field1, type1, field2, type2}``` and ```parts = {{field1, type1, ...}, {field2, type2, ...}}```. In the example the format is guessed as the first variant and an extra option is ignored. We should produce an error or (better) accept one-part definition as ```parts = {field1, type1, ...}```.

### PR DESCRIPTION
**Pull request by mary3000**
Previously accepted formats of index parts:
parts = {field1, type1, field2, type2}, or
parts = {{field1, type1, ...}, {field2, type2, ...}}

Now it is allowed to write without extra brace if there is one part only:
parts = {field1, type1, ...}

Closes #2866

### CODE REVIEW

File: test/box/misc.test.lua [hunk 1]
```diff
@@ -348,3 +348,19 @@ tuple_format = box.internal.new_tuple_format(format)
...
 local ffi = require'ffi' ffi.C._say(ffi.C.S_WARN, nil, 0, nil, "%s", "test log")
 test_run:grep_log('default', 'test log')
+
+--
+-- gh-2866: one more way to declare index parts
+--
+s = box.schema.space.create('test')
+i = s:create_index('test1', {parts = {{1, 'unsigned'}}})
+i = s:create_index('test2', {parts = {{2, 'string', is_nullable = true, collation = 'unicode'}}})
+i.parts[1].is_nullable
```
Comment (alyapunov) [inline test/box/misc.test.lua:358]: As for me, instead of several lines `i = <...> \n i.parts[1].is_nullable \n ... ` it would be better to make one short line `<...>.parts`. It will simply show how part options was parsed.

File: test/box/misc.test.lua [hunk 2]
```diff
@@ -348,3 +348,19 @@ tuple_format = box.internal.new_tuple_format(format)
...
+i.parts[1].collation
+i = s:create_index('test4', {parts = {3, 'string', is_nullable = true}})
```
Comment (alyapunov) [inline test/box/misc.test.lua:360]: I would be glad if there was one more test with similar options but with `collation = 'unicode'`. Just to gurantee that it works with another options too.

**Review by alyapunov — CHANGES_REQUESTED** [review]

File: test/box/misc.test.lua (hunk 1 above)
Comment (mary3000) [reply test/box/misc.test.lua]: Fixed

File: test/box/misc.test.lua (hunk 2 above)
Comment (mary3000) [reply test/box/misc.test.lua:359]: Fixed (see test5)

**Review by Korablev77 — APPROVED** [review]

**Review by alyapunov — APPROVED** [review]
LGTM

**Comment by Korablev77** [comment]
Pushed to master, changelog is updated (2.7).
